client/src/components/UploadCode.js

```diff
@@ -0,0 +1,142 @@
+import React, { useState } from "react";
+// import { makeStyles } from "@material-ui/core/styles";
+import Button from "@material-ui/core/Button";
+import Dialog from "@material-ui/core/Dialog";
+import Grid from "@material-ui/core/Grid";
+// import DialogActions from "@material-ui/core/DialogActions";
+import DialogContent from "@material-ui/core/DialogContent";
+import PageHeader from "../elements/PageHeader";
+import SubmitButton from "../elements/SubmitButton";
+import TextField from "@material-ui/core/TextField";
+import Typography from "@material-ui/core/Typography";
+import PrismEditor from "../components/Editor/DraftEditor";
+import MenuItem from "@material-ui/core/MenuItem";
+import { Box } from "@material-ui/core";
+import Alert from "../elements/SnackBar";
+
+// const useStyles = makeStyles((theme) => ({}));
+
+export default function AddCodeDialog() {
+  // const classes = useStyles();
+  const [open, setOpen] = useState(false);
+  const [title, setTitle] = useState("");
+  const [language, setLanguage] = useState("");
+  const [makeSubmit, setMakeSubmit] = useState(false);
+  const [editorHasContent, setEditorHasContent] = useState(false);
+  const [error, setError] = useState(false);
+  const languages = [
+    "Python",
+    "JavaScript",
+    "Java",
+    "PHP",
+    "GoLang",
+    "C#",
+    "C++",
+    "Ruby",
+  ];
+
+  const handleSubmit = (text) => {
+    //Wrap up editor data with user and language data and send to server.
+    const request = {
+      title: title,
+      language: languages,
+      content: text,
+    };
+    //SEND REQUEST AND close dialog
+    console.log(request);
+    setTitle("");
+    setLanguage("");
+    setOpen(false);
+  };
+
+  const handleHasContent = (value) => {
+    setEditorHasContent(value);
+  };
+
+  const startSubmit = () => {
+    if (!language || !title || !editorHasContent) {
+      setError(true);
+    } else {
+      setMakeSubmit(true);
+    }
+  };
+
+  return (
+    <React.Fragment>
+      <Button
+        variant="outlined"
+        color="primary"
+        onClick={() => {
+          setOpen(!open);
+        }}
+      >
+        Open max-width dialog
+      </Button>
+      <Dialog
+        fullWidth
+        maxWidth="md"
+        open={open}
+        onClose={() => {
+          setOpen(!open);
+        }}
+      >
```

Comment:
This component should probably just be the dialog right? Can you make this component show again only when the navbar button is pressed as opposed to putting it in a testing route?